components/producers/semgrep/main.go

```diff
@@ -72,3 +75,41 @@ func parseIssues(out types.SemgrepResults) ([]*v1.Issue, error) {
 	}
 	return issues, nil
 }
+
+// Semgrep CWEs can be a string or an array of strings
+func handleSemgrepCWE(cwe interface{}) []int32 {
+	cweInts := []int32{}
+
+	switch v := cwe.(type) {
+	case []interface{}:
+		for _, s := range v {
+			cweAsInt := convertCWEStringToInt(s.(string))
+			cweInts = append(cweInts, int32(cweAsInt))
+		}
+	case string:
+		cweAsInt := convertCWEStringToInt(v)
+		cweInts = append(cweInts, int32(cweAsInt))
+	default:
+		log.Fatalf("invalid cwe type: %T", cwe)
```

Comment:
we should not use log.Fatalf anywhere outside of a main, we should return an error instead and let the caller decide if they want to stop executing or keep going. there are a bunch of `log.Fatal` used also further down that should be fixed

Reply:
fixed! thanks